pkg/agent/inteface.go

```diff
@@ -202,14 +212,41 @@ func InstallAllStrategy(installNamespace string) *InstallStrategy {
 	return &InstallStrategy{
 		Type:             InstallAll,
 		InstallNamespace: installNamespace,
+		managedClusterFilter: func(cluster *clusterv1.ManagedCluster) bool {
+			return true
+		},
 	}
 }
 
 func InstallByLabelStrategy(installNamespace string, selector metav1.LabelSelector) *InstallStrategy {
```

Comment:
I think selector should be a pointer

Reply:
Agree, but that may break the backward capability, how about we add a `InstallWithLabel` using pointer and deprecate `InstallByLable`?
Or we just change the `InstallByLabelStrategy ` directly if there is no addon using it now?

I also kindly think `InstallWithFilterFunction` is better than `InstallByFilterFunction`.